data/src/main/scala/ch.epfl.scala.index.data/github/GithubDownload.scala

```diff
@@ -246,49 +257,98 @@ class GithubDownload(paths: DataPaths,
   }
 
   /**
-    * Process the downloaded data from GraphQL API
+    * Process the downloaded data from repository info
     *
     * @param repo the current repo
     * @param response the response
     * @return
     */
+  private def processCommunityProfileResponse(repo: GithubRepo,
+                                  response: WSResponse): Unit = {
+
+    if (200 == response.status) {
+
+      saveJson(githubRepoCommunityProfilePath(paths, repo), repo, response.body)
+    }
+
+    ()
+  }
+
+  /**
+    * Process the downloaded data from GraphQL API
+    * @param isRetry whether or not the call to this function is a retry
+    * @param repo the current repo
+    * @param response the response
+    * @return
+    */
   private def processTopicsResponse(isRetry: Boolean)(
       repo: GithubRepo,
-      response: WSResponse): Unit = {
+      response: WSResponse): Try[Unit] = Try {
+
+      checkGraphqlError[GithubRepo](repo, response, isRetry, topicQuery, processTopicsResponse)
+      saveJson(githubRepoTopicsPath(paths, repo), repo, response.body)
```

Comment:
Haven't tested it yet since I'm also refactoring how the code handles errors from the GraphQL API as part of another issue.

Are you sure `checkGraphqlError` doesn't throw an exception anymore? I didn't change that method. I just changed the methods that call it (`processIssuesResponse` and `processTopicsResponse`) to return a `Try[Unit]`. So then when `checkGraphqlError` throws its error, it gets returned as a `Failure` by the process methods which gets [handled](https://github.com/scalacenter/scaladex/pull/432/commits/0d0e08ce6dbd290c746345aa8c56493a424dcd08#diff-9b0b9f250047a450506bda2259c11537R129) in `downloadGraphql`